src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -32,68 +33,132 @@ static void Main(string[] args)
                    
                     if(_options.Output != null)
                     {
-                        if (!Directory.Exists(_options.Output))
+                        if (Path.GetExtension(_options.Output) != ".txt")
+                        {
+                            Console.Error.WriteLine("Error: output must be into a *.txt file");
+                            Environment.Exit(-3);
+                        }
+
+                        string directory = _options.Output.Substring(0, _options.Output.LastIndexOf("/", StringComparison.OrdinalIgnoreCase));
+
+                        if (!Directory.Exists(directory))
                         {
-                            Directory.CreateDirectory(_options.Output);
+                            Directory.CreateDirectory(directory);
+                        }
+
+                        if (!File.Exists(_options.Output))
+                        {
+                            using (FileStream stream = File.Create(_options.Output)) { }
                         }
                     }
 
                     if (_options.Level != null)
                     {
+                        StreamWriter writer =null;
                         try
                         {
                             level = (Level)Enum.Parse(typeof(Level), _options.Level, true);
-                            WriteTestDllAndName("file", _options.Output, _options.DLLs, level);
+
+                            if (_options.Output == null)
+                            {
+                                writer = new StreamWriter(Console.OpenStandardOutput());
+                            }
+                            else
+                            {
+                                writer = new StreamWriter(_options.Output, true);
```

Comment:
Without looking at the MSDN, it's not obvious what the true here does.  You should use this instead:
`writer = new StreamWriter(_options.Output, append: true);`